frontend/app/Editor/Editor.js

```diff
@@ -87,7 +87,11 @@ const peerExtension = (socket, filename, startVersion) => {
             while (!this.done) {
                 const version = getSyncedVersion(this.view.state);
                 const updates = await pullUpdates(socket, filename, version);
-                this.view.dispatch(receiveUpdates(this.view.state, updates));
+                try {
```

Comment:
It is good to have error management, however more preventative measures have to be taken.